src/server/ua_subscription.c

```diff
@@ -439,9 +439,13 @@ UA_Subscription_publish(UA_Server *server, UA_Subscription *sub) {
 
     /* Sample the MonitoredItems with sampling interval <0 (which implies
      * sampling in the same interval as the subscription) */
-    UA_MonitoredItem *mon;
-    LIST_FOREACH(mon, &sub->samplingMonitoredItems, sampling.samplingListEntry) {
-        monitoredItem_sampleCallback(server, mon);
+    UA_DateTime currentTime = UA_DateTime_nowMonotonic();
+    if (sub->lastSamplingListEntry + (UA_DateTime)(sub->publishingInterval * UA_DATETIME_MSEC) <= currentTime) {
```

Comment:
If the function UA_Subscription_publish is called with the publishinginterval of the subscriber, a small jitter of the calling context might result in the issue that in one cycle the monitored items are not sampled, because the entire interval was not reached. Is there a possibility to fix this?

Reply:
hi @keba-estr, Do you have an idea to fix it?

Reply:
by using calculateNextTime ?

Reply:
Yes, calculateNextTime seems to handle this issue correctly by getting the time without any delay in the current cycle.

Reply:
Second parameter of calculateNextTime is used incorrectly. Should be the last time.

Reply:
calculateNextTime is currently declared only local in ua_timer.c.

Reply:
ok fixed. thanks for the quick support 👍